### DISCUSSION TITLE
capping capacity factor for fossil plants

### DISCUSSION THREAD

**Discussion by ssattler** [NREL/ReEDS-2.0#150] [Q&A: Analysis] [answered]
I'd like to cap the capacity factor of coal plants in different years (varying the cap for a period of time), and would also like to add in different gas CF caps as well. Any potential problems with this approach? Any advice on how I might do this? Thanks!

**Comment by wesleyjcole** [comment, marked as answer]
There shouldn't be any problems with this approach.  You can replicate eq_min_cf in the model and turn it into eq_max_cf by just flipping the greater than to a less than.

**Reply by ssattler** [reply]
Thank you! Could I get more help with this? I would like to cap the capacity factor of the coal fleet by a certain year. I've found the code in c_supplymodel but it looks like it's set up for all techs over all years. How do I isolate just the coal techs and set the cap for one year?

*requirement for fleet of a given tech to have a minimum annual capacity factor
eq_min_cf(i,transreg,t)$[minCF(i,t)$tmodel(t)$sum{(v,r)$r_transreg(r,transreg), valgen(i,v,r,t) }$Sw_MinCF]..

    sum{(v,r,h)$[valgen(i,v,r,t)$r_transreg(r,transreg)], hours(h) * GEN(i,v,r,h,t) }

    =g=

    sum{(v,r)$r_transreg(r,transreg), sum{rr$[cap_agg(r,rr)$valcap(i,v,rr,t)], CAP(i,v,rr,t) } }
    * sum{h, hours(h) } * minCF(i,t)
;

**Reply by wesleyjcole** [reply]
You can change it to only apply to coal techs by just adding the coal tech filter (`$coal(i)`).  If you only want it to apply for one year, you can use the yeart(t) set: `$(yeart(t) = 2030)`.

**Reply by ssattler** [reply]
Thanks! It looks like the coal(i) set includes CCS as well as normal coal. How do I identify a set with coal without CCS?

**Reply by wesleyjcole** [reply]
You can use as many conditionals as you want.  So you could do `$coal(i)$(not ccs(i))`.

**Reply by ssattler** [reply]
This is getting an error when I run the model. I declared a new eq_max_cf variable in the Supply Side Variable Declaration right after eq_min_cf and then added the lines below after the eq_min_cf requirement:

eq_max_cf(i,transreg,t)$[coal(i)$(not ccs(i))$(yeart(t) >= 2030)]$[minCF(i,t)$tmodel(t)$sum{(v,r)$r_transreg(r,transreg), valgen(i,v,r,t) }$Sw_MinCF]..

    sum{(v,r,h)$[valgen(i,v,r,t)$r_transreg(r,transreg)], hours(h) * GEN(i,v,r,h,t) }

    =l=

    0.2
;

I'm trying to limit the coal fleet (not CCS) to a 20% CF after 2030.
I get the following error:

1710212  eq_min_cf(i,transreg,t)$[coal(i)$(not ccs(i))$(yeart(t) >= 2030)]$[minCF(i,t)$tmodel(t)$sum{(v,r)$r_transreg(r,transreg), valgen(i,v,r,t) }$Sw_MinCF]..
****             $150                                                     $36                                                                                  $409
**** LINE   1168 INCLUDE     C:\ReEDS_OpenAccess-MB_IRA\ReEDS_OpenAccess-MB_IRA\runs\coalCF2_7_16_Mid_Case\c_supplymodel.gms
**** LINE      2 INPUT       C:\ReEDS_OpenAccess-MB_IRA\ReEDS_OpenAccess-MB_IRA\runs\coalCF2_7_16_Mid_Case\createmodel.gms
1710213   
1710214      sum{(v,r,h)$[valgen(i,v,r,t)$r_transreg(r,transreg)], hours(h) * GEN(i,v,r,h,t) }
1710215   
1710216      =l=
1710217   
1710218      0.2
1710219  ;

Can you help?

**Reply by wesleyjcole** [reply]
If you scroll down near the bottom of the .lst file, it will show you what the errors 150, 36, and 409 mean.  Looking at those error will hopefully help you see where the issue is.  Just looking at your code, you have two sets of brackets in the conditional.  Collapsing it to one set of brackets might fix the error, but I can't say for sure.
